azurerm/internal/services/databricks/databricks_workspace_resource.go

```diff
@@ -79,46 +79,98 @@ func resourceDatabricksWorkspace() *pluginsdk.Resource {
 				MaxItems: 1,
 				Elem: &pluginsdk.Resource{
 					Schema: map[string]*pluginsdk.Schema{
-						"no_public_ip": {
-							Type:     pluginsdk.TypeBool,
-							ForceNew: true,
+						"aml_workspace_id": {
+							Type:         pluginsdk.TypeString,
+							ForceNew:     true,
+							Optional:     true,
+							ValidateFunc: azure.ValidateResourceID,
+							AtLeastOneOf: workspaceCustomParametersString(),
+						},
+
+						"customer_managed_key": {
+							Type:     pluginsdk.TypeList,
 							Optional: true,
-							AtLeastOneOf: []string{"custom_parameters.0.no_public_ip", "custom_parameters.0.public_subnet_name",
-								"custom_parameters.0.private_subnet_name", "custom_parameters.0.virtual_network_id",
+							Computed: true,
+							MaxItems: 1,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"key_source": {
+										Type:             pluginsdk.TypeString,
+										Optional:         true,
+										Default:          "Default",
+										DiffSuppressFunc: suppress.CaseDifference,
+										ValidateFunc: validation.StringInSlice([]string{
+											string(databricks.Default),
+											string(databricks.MicrosoftKeyvault),
+										}, true),
+									},
+									"key_name": {
+										Type:     pluginsdk.TypeString,
+										Optional: true,
+									},
+									"key_version": {
+										Type:     pluginsdk.TypeString,
+										Optional: true,
+									},
+									"key_vault_uri": {
+										Type:     pluginsdk.TypeString,
+										Optional: true,
+									},
+								},
 							},
+							AtLeastOneOf: workspaceCustomParametersString(),
+						},
+
+						"no_public_ip": {
```

Comment:
```suggestion
						"public_ip_disabled": {
```

Reply:
I can't change this as it has already shipped as [no_public_ip](https://github.com/terraform-providers/terraform-provider-azurerm/blob/2b36b7bfb90cd04c135cd76ef7631b15fd5921f4/azurerm/internal/services/databricks/databricks_workspace_resource.go#L82). I can add a TODO to remove this and rename it in 3.0 if you'd like?